src/Service.Tests/SqlTests/RestApiTests/Put/PutApiTestBase.cs

```diff
@@ -118,6 +118,83 @@ public virtual async Task PutOne_Update_Test()
                 );
         }
 
+        /// <summary>
+        /// Tests the PutOne functionality with a REST PUT request.
+        /// Also validates that an empty response body is returned since the role has
+        /// no read action configured
+        /// </summary>
+        [TestMethod]
+        public virtual async Task PutOne_Update_WithNoReadAction_Test()
+        {
+            string requestBody = @"
+            {
+                ""title"": ""The Hobbit Returns to The Shire"",
+                ""publisher_id"": 1234
+            }";
+
+            await SetupAndRunRestApiTest(
+                    primaryKeyRoute: "id/7",
+                    queryString: null,
+                    entityNameOrPath: _integrationEntityName,
+                    sqlQuery: GetQuery(nameof(PutOne_Update_WithNoReadAction_Test)),
+                    operationType: EntityActionOperation.Upsert,
+                    requestBody: requestBody,
+                    expectedStatusCode: HttpStatusCode.OK,
+                    clientRoleHeader: "policy_tester_noread"
+                );
+        }
+
+        /// <summary>
+        /// Tests the PutOne functionality with a REST PUT request.
+        /// This test also validates that the response body contains fields in accordance to the
+        /// fields configured for the read action for the role.
+        /// </summary>
+        [TestMethod]
+        public virtual async Task PutOne_Update_WithExcludeFields_Test()
+        {
+            string requestBody = @"
+            {
+                ""title"": ""The Hobbit Returns to The Shire"",
+                ""publisher_id"": 1234
+            }";
+
+            await SetupAndRunRestApiTest(
+                    primaryKeyRoute: "id/7",
+                    queryString: null,
+                    entityNameOrPath: _integrationEntityName,
+                    sqlQuery: GetQuery(nameof(PutOne_Update_WithExcludeFields_Test)),
+                    operationType: EntityActionOperation.Upsert,
+                    requestBody: requestBody,
+                    expectedStatusCode: HttpStatusCode.OK,
+                    clientRoleHeader: "policy_tester_excludefields"
+                );
+        }
+
+        /// <summary>
+        /// Tests the PutOne functionality with a REST PUT request.
+        /// This test also validates that the database policy configured for the read action is honored.
```

Comment:
nit: add details about the exact database policy configured for read and how it contributes to an empty response